transport/azure/azure.go

```diff
@@ -32,51 +44,119 @@ func New() (http.RoundTripper, error) {
 		return nil, err
 	}
 
-	serviceURL := azblob.NewServiceURL(*endpointURL, pipeline)
+	if config.ABSKey != "" {
+		sharedKeyCredential, err = azblob.NewSharedKeyCredential(config.ABSName, config.ABSKey)
+		if err != nil {
+			return nil, err
+		}
 
-	return transport{&serviceURL}, nil
-}
+		client, err = azblob.NewClientWithSharedKeyCredential(endpointURL.String(), sharedKeyCredential, nil)
+	} else {
+		defaultAzureCredential, err = azidentity.NewDefaultAzureCredential(nil)
+		if err != nil {
+			return nil, err
+		}
 
-func (t transport) RoundTrip(req *http.Request) (resp *http.Response, err error) {
-	containerURL := t.serviceURL.NewContainerURL(strings.ToLower(req.URL.Host))
-	blobURL := containerURL.NewBlockBlobURL(strings.TrimPrefix(req.URL.Path, "/"))
+		client, err = azblob.NewClient(endpointURL.String(), defaultAzureCredential, nil)
+	}
 
-	start, end, err := httprange.Parse(req.Header.Get("Range"))
 	if err != nil {
-		return httprange.InvalidHTTPRangeResponse(req), nil
+		return nil, err
 	}
 
-	length := end - start + 1
-	if end <= 0 {
-		length = azblob.CountToEnd
+	return transport{client}, nil
+}
+
+func (t transport) RoundTrip(req *http.Request) (*http.Response, error) {
+	container := req.URL.Host
+	key := req.URL.Path
+
+	var statusCode = http.StatusOK
+
+	header := make(http.Header)
+	opts := &blob.DownloadStreamOptions{}
+
+	if r := req.Header.Get("Range"); len(r) != 0 {
+		start, end, err := httprange.Parse(r)
+		if err != nil {
+			return httprange.InvalidHTTPRangeResponse(req), err
+		}
+
+		if end != 0 {
+			length := end - start + 1
+			if end < 0 {
+				length = -1
```

Comment:
While it looks like `-1` should work here, I'd use `blob.CountToEnd`

Reply:
Unfortunately, under the newer azblob package, that property no longer exists in `azblob`.

Reply:
It appears that this constant has been moved to `blockblob`, I am making the appropriate refactor now.